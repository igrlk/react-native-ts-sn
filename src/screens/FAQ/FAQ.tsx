import React, { useEffect, useRef } from 'react';
import { View, Text, ScrollView, TouchableOpacity } from 'react-native';

import Screen from 'library/common/commonComponents/Screen';
import GoBackButton from 'library/common/commonComponents/Buttons/GoBackButton';
import { NavigationParams } from 'library/common/commonTypes/navigation';
import Vector from 'resources/images/svg/Vector';

import styles from './FAQStyles';

interface FAQProps {
	navigation: NavigationParams;
}

export default function FAQ({ navigation }: FAQProps) {
	const params = navigation.state.params || {};
	const scrollRef = useRef<ScrollView>(null);

	useEffect(() => {
		if (params.toSamples && scrollRef.current) {
			scrollRef.current.scrollTo(2270);
		}
	}, [navigation]);

	const goToTop = () => {
		if (scrollRef.current) {
			scrollRef.current.scrollTo({x: 0, y: 0, animated: true});
		}
	}

	return (
		<Screen scrollRef={scrollRef}>
			<GoBackButton>FAQ</GoBackButton>

			<View style={styles.container}>
				{faqItems.map((faqItem, index) => (
					<View key={index}>
						<Text style={[styles.text, styles.title]}>{faqItem.title}</Text>
						<Text style={styles.text}>{faqItem.text}</Text>
					</View>
				))}
			</View>

			<TouchableOpacity style={styles.button} onPress={goToTop}>
				<Vector />
			</TouchableOpacity>
		</Screen>
	);
}

export const faqItems = [
	{
		title: 'What is Sentinel?',
		text:
			'Sentinel is a cutting-edge diagnostic service that can detect diseases before visual symptoms occur and screens for over 1,000 pathogens in a single test. This gives growers the ability to make data-driven decisions about their crops. We pride ourselves on providing fast, accurate, affordable diagnostics.',
	},
	{
		title: 'What is the Sentinel diagnostic process?',
		text:
			'When you request services, we will ship you a sampling kit that includes instructions and a pre-paid return mailing envelope. You will sample your plants according to the directions and ship us back the leaves. Then we do all the hard work to diagnose your plants and return the results in a simple, easily shared format in the app.',
	},
	{
		title: 'How do we know Sentinel works?',
		text:
			'We have been using this process to diagnose plants for over three years now and it has been extremely successful in vegetable crops. We have done sensitivity testing with bacteria and viruses to ensure our detection levels are sufficient and have performed a pressure-test to determine how many different pathogens we can detect in a single sample. We are now working with beta testers to ensure that the process works with other types of crops such as berries, grapes, stone fruit, and nut trees.',
	},
	{
		title: 'What diseases does this service test for?',
		text:
			'We test for over 1,000 plant diseases simultaneously. That includes bacteria, fungi, oomycetes, viruses, and viroids. In the future, the full list will be available for you to download.',
	},
	{
		title: 'How can you test for so many diseases at once?',
		text:
			'We took a diagnostic method from human healthcare and have adapted it for agriculture. We use genetic information from the plant combined with a proprietary data analysis process to ensure we are capturing every possible pathogen in your sample.',
	},
	{
		title: 'On what crops can you perform Sentinel testing?',
		text:
			'Currently, we know that Sentinel works on vegetable crops including tomato, pepper, melon, watermelon, cucumber, squash, lettuce, broccoli, spinach, and corn. We are completing beta testing to show that Sentinel also works on grape, berries, nut trees, and fruit trees.',
	},
	{
		title: 'What is in the sampling kit?',
		text:
			'The sampling kit contains a detailed instruction page with how to collect leaves and the size of leaf required, resealable plastic baggies to collect leaves in, and a pre-paid return mailing envelope for you to ship the samples back to us.',
		isForSamples: true,
	},
	{
		title: 'What is a sample?',
		text:
			'A single sample can contain anywhere from 1 to 100 leaves. If you are seeing symptoms in your field, you can send us symptomatic leaves from just a few plants to diagnose. If you want to do a broad screen for pathogens, you can send us up to 100 randomly sampled leaves from your growing location for us to test.',
	},
	{
		title: 'How many samples can I get tested at one time?',
		text:
			'Beta testers may submit up to 5 samples for free during this phase of our product testing and launch.',
	},
	{
		title: 'Where do I sample off the plant?',
		text:
			'We want the youngest leaf from the plant that is the correct size. You can see an example outline of the leaf size required on the sampling instruction page.',
	},
	{
		title: 'How big of a leaf should I pick?',
		text:
			'The sampling instructions page has an outline of a leaf to give you an idea of the leaf size requested. In general, the leaf needs to be at least the size of a quarter.',
	},
	{
		title: 'Do I need to ship the leaves back on ice?',
		text:
			'No. We provide you with an overnight shipping label so that you can simply put the leaves in the sampling bag and ship it back to us.',
	},
	{
		title: 'How long will it take to get results once I ship my sample?',
		text:
			'You can expect results in the app in 10-14 business days from the time your samples are received in our lab.',
	},
	{
		title: 'How can I contact the Sentinel Team if I have a problem?',
		text:
			'Email: contact@sentinelagdiagnostics.com\nPhone: 1-530-669-6223',
	},
];
