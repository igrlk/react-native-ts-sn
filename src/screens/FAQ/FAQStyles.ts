import EStyleSheet from 'react-native-extended-stylesheet';

export default EStyleSheet.create({
	container: {
		paddingTop: 20,
	},
	title: {
		fontWeight: 'bold',
		fontSize: '$fontSize.medium',
		paddingBottom: 0,
	},
	text: {
		fontFamily: 'Poppins',
		fontSize: '$fontSize.mini',
		paddingBottom: 20,
	},
	button: {
		display: 'flex',
		alignItems: 'flex-end',
		marginBottom: 40,
	},
});
