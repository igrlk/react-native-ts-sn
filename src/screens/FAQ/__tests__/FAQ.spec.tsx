import React from 'react';
import renderer from 'react-test-renderer';

import FAQ from '../FAQ';

describe('FAQ', () => {
	it('match to snapshot', () => {
		const tree = renderer.create(<FAQ navigation={{ state: { params: {} } } as any} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
