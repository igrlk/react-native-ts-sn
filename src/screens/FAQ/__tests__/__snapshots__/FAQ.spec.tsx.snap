// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FAQ match to snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        undefined,
        undefined,
        Object {},
      ]
    }
  >
    <View>
      <RCTScrollView>
        <View>
          <View>
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight="22"
                bbWidth="12"
                fill="none"
                focusable={false}
                height="22"
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    undefined,
                    Object {
                      "opacity": 1,
                    },
                    Object {
                      "flex": 0,
                      "height": 22,
                      "width": 12,
                    },
                  ]
                }
                vbHeight={22}
                vbWidth={12}
                width="12"
              >
                <RNSVGGroup
                  fill={null}
                  fillOpacity={1}
                  fillRule={1}
                  font={Object {}}
                  matrix={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                  opacity={1}
                  originX={0}
                  originY={0}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                  rotation={0}
                  scaleX={1}
                  scaleY={1}
                  skewX={0}
                  skewY={0}
                  stroke={null}
                  strokeDasharray={null}
                  strokeDashoffset={null}
                  strokeLinecap={0}
                  strokeLinejoin={0}
                  strokeMiterlimit={4}
                  strokeOpacity={1}
                  strokeWidth={1}
                  vectorEffect={0}
                  x={0}
                  y={0}
                >
                  <RNSVGPath
                    d="M1.69385 10.9994L11.6294 1.06387C11.8738 0.819511 11.8738 0.427621 11.6294 0.183266C11.3851 -0.0610887 10.9932 -0.0610887 10.7488 0.183266L0.370644 10.5614C0.126289 10.8058 0.126289 11.1977 0.370644 11.442L10.7488 21.8156C10.8687 21.9355 11.0301 22 11.1868 22C11.3436 22 11.5049 21.9401 11.6248 21.8156C11.8692 21.5712 11.8692 21.1793 11.6248 20.935L1.69385 10.9994Z"
                    fill={
                      Array [
                        0,
                        4278190080,
                      ]
                    }
                    fillOpacity={1}
                    fillRule={1}
                    matrix={
                      Array [
                        1,
                        0,
                        0,
                        1,
                        0,
                        0,
                      ]
                    }
                    opacity={1}
                    originX={0}
                    originY={0}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                    rotation={0}
                    scaleX={1}
                    scaleY={1}
                    skewX={0}
                    skewY={0}
                    stroke={null}
                    strokeDasharray={null}
                    strokeDashoffset={null}
                    strokeLinecap={0}
                    strokeLinejoin={0}
                    strokeMiterlimit={4}
                    strokeOpacity={1}
                    strokeWidth={1}
                    vectorEffect={0}
                    x={0}
                    y={0}
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
            <Text>
              FAQ
            </Text>
          </View>
          <View>
            <View>
              <Text
                style={
                  Array [
                    undefined,
                    undefined,
                  ]
                }
              >
                What is Sentinel?
              </Text>
              <Text>
                Sentinel is a cutting-edge diagnostic service that can detect diseases before visual symptoms occur and screens for over 1,000 pathogens in a single test. This gives growers the ability to make data-driven decisions about their crops. We pride ourselves on providing fast, accurate, affordable diagnostics.
              </Text>
            </View>
            <View>
              <Text
                style={
                  Array [
                    undefined,
                    undefined,
                  ]
                }
              >
                What is the Sentinel diagnostic process?
              </Text>
              <Text>
                When you request services, we will ship you a sampling kit that includes instructions and a pre-paid return mailing envelope. You will sample your plants according to the directions and ship us back the leaves. Then we do all the hard work to diagnose your plants and return the results in a simple, easily shared format in the app.
              </Text>
            </View>
            <View>
              <Text
                style={
                  Array [
                    undefined,
                    undefined,
                  ]
                }
              >
                How do we know Sentinel works?
              </Text>
              <Text>
                We have been using this process to diagnose plants for over three years now and it has been extremely successful in vegetable crops. We have done sensitivity testing with bacteria and viruses to ensure our detection levels are sufficient and have performed a pressure-test to determine how many different pathogens we can detect in a single sample. We are now working with beta testers to ensure that the process works with other types of crops such as berries, grapes, stone fruit, and nut trees.
              </Text>
            </View>
            <View>
              <Text
                style={
                  Array [
                    undefined,
                    undefined,
                  ]
                }
              >
                What diseases does this service test for?
              </Text>
              <Text>
                We test for over 1,000 plant diseases simultaneously. That includes bacteria, fungi, oomycetes, viruses, and viroids. In the future, the full list will be available for you to download.
              </Text>
            </View>
            <View>
              <Text
                style={
                  Array [
                    undefined,
                    undefined,
                  ]
                }
              >
                How can you test for so many diseases at once?
              </Text>
              <Text>
                We took a diagnostic method from human healthcare and have adapted it for agriculture. We use genetic information from the plant combined with a proprietary data analysis process to ensure we are capturing every possible pathogen in your sample.
              </Text>
            </View>
            <View>
              <Text
                style={
                  Array [
                    undefined,
                    undefined,
                  ]
                }
              >
                On what crops can you perform Sentinel testing?
              </Text>
              <Text>
                Currently, we know that Sentinel works on vegetable crops including tomato, pepper, melon, watermelon, cucumber, squash, lettuce, broccoli, spinach, and corn. We are completing beta testing to show that Sentinel also works on grape, berries, nut trees, and fruit trees.
              </Text>
            </View>
            <View>
              <Text
                style={
                  Array [
                    undefined,
                    undefined,
                  ]
                }
              >
                What is in the sampling kit?
              </Text>
              <Text>
                The sampling kit contains a detailed instruction page with how to collect leaves and the size of leaf required, resealable plastic baggies to collect leaves in, and a pre-paid return mailing envelope for you to ship the samples back to us.
              </Text>
            </View>
            <View>
              <Text
                style={
                  Array [
                    undefined,
                    undefined,
                  ]
                }
              >
                What is a sample?
              </Text>
              <Text>
                A single sample can contain anywhere from 1 to 100 leaves. If you are seeing symptoms in your field, you can send us symptomatic leaves from just a few plants to diagnose. If you want to do a broad screen for pathogens, you can send us up to 100 randomly sampled leaves from your growing location for us to test.
              </Text>
            </View>
            <View>
              <Text
                style={
                  Array [
                    undefined,
                    undefined,
                  ]
                }
              >
                How many samples can I get tested at one time?
              </Text>
              <Text>
                Beta testers may submit up to 5 samples for free during this phase of our product testing and launch.
              </Text>
            </View>
            <View>
              <Text
                style={
                  Array [
                    undefined,
                    undefined,
                  ]
                }
              >
                Where do I sample off the plant?
              </Text>
              <Text>
                We want the youngest leaf from the plant that is the correct size. You can see an example outline of the leaf size required on the sampling instruction page.
              </Text>
            </View>
            <View>
              <Text
                style={
                  Array [
                    undefined,
                    undefined,
                  ]
                }
              >
                How big of a leaf should I pick?
              </Text>
              <Text>
                The sampling instructions page has an outline of a leaf to give you an idea of the leaf size requested. In general, the leaf needs to be at least the size of a quarter.
              </Text>
            </View>
            <View>
              <Text
                style={
                  Array [
                    undefined,
                    undefined,
                  ]
                }
              >
                Do I need to ship the leaves back on ice?
              </Text>
              <Text>
                No. We provide you with an overnight shipping label so that you can simply put the leaves in the sampling bag and ship it back to us.
              </Text>
            </View>
            <View>
              <Text
                style={
                  Array [
                    undefined,
                    undefined,
                  ]
                }
              >
                How long will it take to get results once I ship my sample?
              </Text>
              <Text>
                You can expect results in the app in 10-14 business days from the time your samples are received in our lab.
              </Text>
            </View>
            <View>
              <Text
                style={
                  Array [
                    undefined,
                    undefined,
                  ]
                }
              >
                How can I contact the Sentinel Team if I have a problem?
              </Text>
              <Text>
                Email: contact@sentinelagdiagnostics.com
Phone: 1-530-669-6223
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight="40"
              bbWidth="40"
              fill="none"
              focusable={false}
              height="40"
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  undefined,
                  Object {
                    "opacity": 1,
                  },
                  Object {
                    "flex": 0,
                    "height": 40,
                    "width": 40,
                  },
                ]
              }
              vbHeight={40}
              vbWidth={40}
              width="40"
            >
              <RNSVGGroup
                fill={null}
                fillOpacity={1}
                fillRule={1}
                font={Object {}}
                matrix={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    0,
                    0,
                  ]
                }
                opacity={1}
                originX={0}
                originY={0}
                propList={
                  Array [
                    "fill",
                  ]
                }
                rotation={0}
                scaleX={1}
                scaleY={1}
                skewX={0}
                skewY={0}
                stroke={null}
                strokeDasharray={null}
                strokeDashoffset={null}
                strokeLinecap={0}
                strokeLinejoin={0}
                strokeMiterlimit={4}
                strokeOpacity={1}
                strokeWidth={1}
                vectorEffect={0}
                x={0}
                y={0}
              >
                <RNSVGPath
                  d="M37.3183 9.96076C35.5298 6.89672 33.1038 4.47072 30.0397 2.68231C26.975 0.893981 23.6295 0 20.0006 0C16.372 0 13.0255 0.893981 9.96139 2.68231C6.89709 4.47045 4.47109 6.89645 2.68258 9.96076C0.893981 13.0252 0 16.3718 0 20C0 23.6285 0.894254 26.9745 2.68231 30.0392C4.47072 33.1032 6.89672 35.5292 9.96103 37.3178C13.0255 39.106 16.3718 40.0001 20.0002 40.0001C23.6287 40.0001 26.9753 39.106 30.0396 37.3178C33.1037 35.5298 35.5296 33.1032 37.318 30.0392C39.106 26.9748 39.9999 23.6283 39.9999 20C40 16.3714 39.106 13.0248 37.3183 9.96076ZM32.9955 23.6714L30.3391 26.3283C30.0088 26.658 29.6184 26.8231 29.1672 26.8231C28.7159 26.8231 28.3248 26.658 27.9954 26.3283L20.0002 18.3332L12.0058 26.3283C11.6759 26.658 11.2854 26.8231 10.8339 26.8231C10.3824 26.8231 9.99177 26.658 9.66185 26.3283L7.00572 23.6714C6.6758 23.3424 6.5108 22.951 6.5108 22.4996C6.5108 22.0488 6.6758 21.6579 7.00572 21.3277L18.8288 9.50496C19.1587 9.17505 19.5492 9.01022 20.0006 9.01022C20.452 9.01022 20.8432 9.17505 21.1723 9.50496L32.9955 21.3277C33.325 21.6579 33.4902 22.0487 33.4902 22.4996C33.4902 22.9509 33.3254 23.3423 32.9955 23.6714Z"
                  fill={
                    Array [
                      0,
                      4282103114,
                    ]
                  }
                  fillOpacity={1}
                  fillRule={1}
                  matrix={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                  opacity={1}
                  originX={0}
                  originY={0}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                  rotation={0}
                  scaleX={1}
                  scaleY={1}
                  skewX={0}
                  skewY={0}
                  stroke={null}
                  strokeDasharray={null}
                  strokeDashoffset={null}
                  strokeLinecap={0}
                  strokeLinejoin={0}
                  strokeMiterlimit={4}
                  strokeOpacity={1}
                  strokeWidth={1}
                  vectorEffect={0}
                  x={0}
                  y={0}
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</View>
`;
